### ISSUE TITLE
Attack Overview with Properties

### ISSUE THREAD

**Issue by jangop** [bethgelab/foolbox#653]
The table at [foolbox.attacks](https://foolbox.readthedocs.io/en/stable/modules/attacks.html#module-foolbox.attacks) is currently the closest thing to an overview of all implemented attacks. It could be an ideal place to list major defining properties, such as whether an attack minimizes perturbations and if it supports targeted attacks (or rather, which criteria it supports).

1. Is it correct that no list with more detail exists?
2. What other properties should be included?
3. Could/should the overview be generated automatically, possibly by inspecting which classes (such as [`MinimizationAttack`](https://github.com/bethgelab/foolbox/blob/92af8673ee957e5af1f9ad5f3abb474c63f93397/foolbox/attacks/base.py#L334)) a given attack inherits from?

**Comment by zimmerrol** [comment]
1. Yes, this is correct.
2. I cannot think of any other properties at the moment - but that shouldn't mean that there aren't any. So I'm open to suggestions here:)
3. Yes, it definitely should be generated automatically since we otherwise have the risk of inconsistencies across different attacks.

@jangop Do you think you can prepare a draft/PR for your suggestion? If so, I would assign the issue to yourself.

**Comment by zimmerrol** [comment]
Is there an update on the status of this idea?

**Comment by jangop** [comment]
Yes. `Update #1`: I have absolutely no idea what I'm doing with `sphinx`, so it will result in an extra script that will spit out a table to be manually included in the documentation.

**Comment by kanak000** [comment]
> 1. Yes, this is correct.
> 
>     2. I cannot think of any other properties at the moment - but that shouldn't mean that there aren't any. So I'm open to suggestions here:)
> 
>     3. Yes, it definitely should be generated automatically since we otherwise have the risk of inconsistencies across different attacks.
> 
> 
> @jangop Do you think you can prepare a draft/PR for your suggestion? If so, I would assign the issue to you.

hey, great to see that you were active here in past. Is it possible for you to post one proper Boundary Attack code from foolbox here. If you want i can help you, I am getting some errors while running it
